sql/sql_class.cc

```diff
@@ -5191,6 +5191,27 @@ thd_rpl_deadlock_check(MYSQL_THD thd, MYSQL_THD other_thd)
     releasing the lock for this transaction so replication can proceed.
   */
 #ifdef HAVE_REPLICATION
+  if (unlikely(opt_slave_parallel_print_all_deadlocks) &&
+      (other_rgi->trans_retries >= 1 ||
+       opt_slave_parallel_print_all_deadlocks == 2))
+  {
+    char thd_info[400];
+    char other_thd_info[400];
+    thd_get_error_context_description(thd, thd_info, sizeof(thd_info), 350);
+    thd_get_error_context_description(other_thd, other_thd_info,
+                                      sizeof(other_thd_info), 350);
+    sql_print_information("Slave SQL thread: Deadlock detected, aborting "
```

Comment:
Adding identifiers of the failed event and a record that either was not found, or locked would certainly be useful.
Sure the failed event part is trivial. To find the key whose searching for ended up in timeout is simple at least for ROW format. Perhaps we could be content with that, unless a way can be found to extract that info in the STATEMENT format.